halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/ArtifactService.java

```diff
@@ -86,10 +89,70 @@ public String getArtifactVersion(String deploymentName, SpinnakerArtifact artifa
     return getBillOfMaterials(deploymentName).getServices().getArtifactVersion(artifact);
   }
 
-  public void writeArtifactConfig(SpinnakerArtifact artifact, String profileName, String contents) {
+  public void writeBom(String bomPath) {
     if (writeableProfileRegistry == null) {
       throw new HalException(new ConfigProblemBuilder(FATAL,
           "You need to set the \"spinnaker.config.input.writer\" property to \"true\" to modify base-profiles.").build());
     }
+
+    BillOfMaterials bom;
+    String bomContents;
+    String version;
+
+    try {
+      bomContents = IOUtils.toString(new FileInputStream(bomPath));
+      bom = strictObjectMapper.convertValue(
+          yaml.load(bomContents),
+          BillOfMaterials.class);
+      version = bom.getVersion();
+    } catch (IOException e) {
+      throw new HalException(new ConfigProblemBuilder(FATAL,
+          "Unable to load Bill of Materials: " + e.getMessage()).build()
+      );
+    }
+
+    if (version == null) {
+      throw new HalException(new ConfigProblemBuilder(FATAL, "No version was supplied in this BOM.").build());
+    }
+
+    writeableProfileRegistry.writeBom(bom.getVersion(), bomContents);
+  }
+
+  public void writeArtifactConfig(String bomPath, String artifactName, String profilePath) {
+    if (writeableProfileRegistry == null) {
+      throw new HalException(new ConfigProblemBuilder(FATAL,
+          "You need to set the \"spinnaker.config.input.writer\" property to \"true\" to modify base-profiles.").build());
```

Comment:
`writer` seems a bit vague to me, I like `writerEnabled`, but po-tay-to po-tah-to.